### PR TITLE
ADEN-5477

### PR DESCRIPTION
**Pull request by v3rth**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/common/templates/TemplateNoFirstLoad.java [hunk 1]
```diff
@@ -11,12 +12,19 @@
...
+    putVASTToLocalStorage(videoAd.getVastXML());
     ads.getUrl(page);
     return ads;
   }
+
+  protected AdsBaseObject openPageWithVideoInLocalStorage(Page page) {
+    return openPageWithVideoInLocalStorage(page, VuapVideos.DEFAULT);
+  }
+
+  private void putVASTToLocalStorage(String vast) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/templates/TemplateNoFirstLoad.java]: MAJOR Remove this unused private "putVASTToLocalStorage" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AutoplayVuap.java#L220](https://github.com/Wikia/selenium-tests/blob/94c909a9253f1418e412704833a440d340dfa173/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java#L220): Remove this unused private "isOverlayNoVisible" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [AdsDataProvider.java#L1268](https://github.com/Wikia/selenium-tests/blob/94c909a9253f1418e412704833a440d340dfa173/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L1268): Move this trailing comment on the previous empty line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java [hunk 2]
```diff
@@ -218,7 +226,7 @@ public void vuapResolvedShouldEnd(Page page, String slot, String videoIframeSele
   }
 
   private void scrollToSlot(String slotName, AdsBaseObject ads) {
-    if (slotName == AdsContent.BOTTOM_LB) {
+    if (slotName.equals(AdsContent.BOTTOM_LB)) {
```
Comment (dianafa) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java]: 💎

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java [hunk 3]
```diff
@@ -177,35 +186,34 @@ public void vuapResolvedStateShouldCloseAfterTapingOnCloseButton(Page page, Stri
...
-      dataProviderClass = AdsDataProvider.class,
-      dataProvider = "adsVuapTngDesktop")
+  @Test(
+    groups = {"AdsVuapResolvedStateTng", "AdsVuapResolvedStateIsNotMuteOasis"},
+    dataProviderClass = AdsDataProvider.class,
+    dataProvider = "adsVuapTngDesktop"
+  )
   public void vuapResolvedStateIsNotMute(Page page, String slot, String videoIframeSelector) {
-    AdsBaseObject ads = openPageWithVideoInLocalStorage(page);
+    AdsBaseObject ads = new AdsBaseObject(driver, page.getUrl() + "&" + TURN_ON_RESOLVED_STATE);
```
Comment (dianafa) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java]: are we sure `&` will be always appropriate here? (not `?`)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/VuapAssertions.java [hunk 4]
```diff
@@ -52,12 +48,15 @@ public static void verifyReplyButtonDisplayedAfterVideoClose(final AutoplayVuap
     vuap.waitForVideoToEnd(maxVideoDuration);
   }
 
-  private static void playVideoForOneSecond(final AutoplayVuap vuap) {
+  private static void playVideoForFewSeconds(final AutoplayVuap vuap) {
     vuap.play();
+
     try {
-      TimeUnit.MILLISECONDS.sleep(200);
+      TimeUnit.SECONDS.sleep(2);
```
Comment (dianafa) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/VuapAssertions.java]: it's not the same amount time, is that on purpose?

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/VideoAd.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+  public String getVastXML() {
+    return String.format(VAST_XML, title, parseDuration(), type, width, height, url);
+  }
+
+  private String parseDuration() {
+    NumberFormat formatter = new DecimalFormat("00");
+    int hours = (int) Math.floor(duration / 3600);
+    int minutes = (int) Math.floor(duration / 60 % 60);
+    int seconds = (int) Math.floor(duration % 60);
```
Comment (dianafa) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/VideoAd.java]: it looks like an universal operation, did we checked if sth similar is not performed in repo already?

Comment (dianafa) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/VideoAd.java]: can we declare those vars using one `int` and commas?

Comment (v3rth) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/VideoAd.java]: I couldn't find anything similar, but you right, it looks like universal

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/VuapAssertions.java (hunk 4 above)
Comment (v3rth) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/VuapAssertions.java]: 2s = 2000ms

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [UrlBuilder.java#L93](https://github.com/Wikia/selenium-tests/blob/346ed11190a0fbbd808406eb8dad6b433f9d7b7b/src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java#L93): Extract this nested code block into a method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1199)
1. ![MAJOR][MAJOR] [UrlBuilder.java#L119](https://github.com/Wikia/selenium-tests/blob/346ed11190a0fbbd808406eb8dad6b433f9d7b7b/src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java#L119): Extract this nested code block into a method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1199)
1. ![MAJOR][MAJOR] [AutoplayVuap.java#L220](https://github.com/Wikia/selenium-tests/blob/346ed11190a0fbbd808406eb8dad6b433f9d7b7b/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java#L220): Remove this unused private "isOverlayNoVisible" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [AdsDataProvider.java#L1268](https://github.com/Wikia/selenium-tests/blob/346ed11190a0fbbd808406eb8dad6b433f9d7b7b/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L1268): Move this trailing comment on the previous empty line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [UrlBuilder.java#L93](https://github.com/Wikia/selenium-tests/blob/390b9a3ce2f86ed61fdde8c449834be1aaa922db/src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java#L93): Extract this nested code block into a method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1199)
1. ![MAJOR][MAJOR] [UrlBuilder.java#L119](https://github.com/Wikia/selenium-tests/blob/390b9a3ce2f86ed61fdde8c449834be1aaa922db/src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java#L119): Extract this nested code block into a method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1199)
1. ![MAJOR][MAJOR] [AutoplayVuap.java#L220](https://github.com/Wikia/selenium-tests/blob/390b9a3ce2f86ed61fdde8c449834be1aaa922db/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java#L220): Remove this unused private "isOverlayNoVisible" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [AdsDataProvider.java#L1268](https://github.com/Wikia/selenium-tests/blob/390b9a3ce2f86ed61fdde8c449834be1aaa922db/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L1268): Move this trailing comment on the previous empty line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 12 issues
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 4 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [UrlBuilder.java#L93](https://github.com/Wikia/selenium-tests/blob/5209ebdabbac2aa8be87309fee28972fa097121e/src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java#L93): Extract this nested code block into a method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1199)
1. ![MAJOR][MAJOR] [UrlBuilder.java#L119](https://github.com/Wikia/selenium-tests/blob/5209ebdabbac2aa8be87309fee28972fa097121e/src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java#L119): Extract this nested code block into a method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1199)
1. ![MAJOR][MAJOR] [AutoplayVuap.java#L220](https://github.com/Wikia/selenium-tests/blob/5209ebdabbac2aa8be87309fee28972fa097121e/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java#L220): Remove this unused private "isOverlayNoVisible" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [MessageWall.java#L44](https://github.com/Wikia/selenium-tests/blob/5209ebdabbac2aa8be87309fee28972fa097121e/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/messagewall/MessageWall.java#L44): Make this final field static too. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1170)
1. ![MAJOR][MAJOR] [MessageWall.java#L46](https://github.com/Wikia/selenium-tests/blob/5209ebdabbac2aa8be87309fee28972fa097121e/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/messagewall/MessageWall.java#L46): Make this final field static too. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1170)
1. ![MAJOR][MAJOR] [MessageWall.java#L47](https://github.com/Wikia/selenium-tests/blob/5209ebdabbac2aa8be87309fee28972fa097121e/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/messagewall/MessageWall.java#L47): Make this final field static too. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1170)
1. ![MINOR][MINOR] [ForumBoardPage.java#L194](https://github.com/Wikia/selenium-tests/blob/5209ebdabbac2aa8be87309fee28972fa097121e/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/forumpageobject/ForumBoardPage.java#L194): Define a constant instead of duplicating this literal ") li.follow" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [ForumBoardPage.java#L194](https://github.com/Wikia/selenium-tests/blob/5209ebdabbac2aa8be87309fee28972fa097121e/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/forumpageobject/ForumBoardPage.java#L194): Define a constant instead of duplicating this literal ".thread:nth-child(" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [ForumBoardPage.java#L198](https://github.com/Wikia/selenium-tests/blob/5209ebdabbac2aa8be87309fee28972fa097121e/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/forumpageobject/ForumBoardPage.java#L198): Define a constant instead of duplicating this literal "unfollowIfDiscussionIsFollowed" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [MessageWall.java#L3](https://github.com/Wikia/selenium-tests/blob/5209ebdabbac2aa8be87309fee28972fa097121e/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/messagewall/MessageWall.java#L3): Remove this unused import 'java.lang.reflect.Type'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**